src/bubble_sort.py

```diff
@@ -0,0 +1,28 @@
+"""Bubble sort algorithm."""
+import timeit
+
+
+def bubble_sort(l):
+    """Bubble Sort: Input list or tuple."""
+    if not isinstance(l, (list, tuple)):
+        raise TypeError('Please input a list or tuple.')
+    n = len(l)
+    swapped = True
+    while swapped:
+        swapped = False
+        for i in range(n - 1):
+            j = i + 1
+            if l[i] > l[j]:
+                temp = l[j]
+                l[j] = l[i]
+                l[i] = temp
```

Comment:
In Python you can do multiple assignment without using some intermediary variable. I imagine that what you wanted was:

```python
l[i], l[j] = l[j], l[i]
```